Source/WebGPU/WebGPU/Queue.swift

```diff
@@ -23,7 +23,32 @@
  * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+private let largeBufferSize = 32 * 1024 * 1024
 
-public func writeBuffer(queue: WebGPU.Queue, buffer: WebGPU.Buffer, offset: UInt64, data: SpanUInt8) {
-    queue.writeBuffer(buffer.buffer(), offset, data)
+public func writeBuffer(
+    queue: WebGPU.Queue, buffer: WebGPU.Buffer, bufferOffset: UInt64, data: SpanUInt8
+) {
+    queue.writeBuffer(mtlBuffer: buffer.buffer(), bufferOffset: bufferOffset, data: data)
+}
+
+extension WebGPU.Queue {
+    public func writeBuffer(mtlBuffer: any MTLBuffer, bufferOffset: UInt64, data: SpanUInt8) {
+        let device = self.device()
+        ensureBlitCommandEncoder()
+        let noCopy = data.size() >= largeBufferSize
+        precondition(device.device() != nil, "WeakPtr to Device is null")
```

Comment:
This may fail, but I don't think we want it to abort. Is `precondition` the correct thing to use in that case?

Reply:
If we assert, the tests validating this in debug would crash.